File: CoffeeShop/Entities/GroupOrder/Order.cs
namespace CoffeeShop.Entities.GroupOrder;

public class Order
{
    public string OrderId { get; private set; }
    public string BuyerId { get; private set; }
    public DateTime CreatedDate { get; private set; }
    public Address ShipAddress { get; private set; }
    public OrderStatus Status { get; private set; }


    private readonly List<OrderItem> _orderItems = new List<OrderItem>();
    public IReadOnlyCollection<OrderItem> OrderItems => _orderItems.AsReadOnly();

    private Order() { }

    public Order(string buyerId, Address shipAddress, OrderStatus status, List<OrderItem> orderItems)
    {
        BuyerId = buyerId;
        ShipAddress = shipAddress;
        Status = status;
        _orderItems = orderItems;
    }

    public decimal Total => _orderItems.Sum(oi => oi.Price * oi.Quantity);
}

File: webcoffee/src/entities/order.ts
export interface Order{
    orderId: number,
    buyerId: number,
    orderDate: Date,
    total: number,
    orderItems: OrderItems[],
    status: string,
    paymentStatus: string,
    address: Address,
    updatedDate: Date
};

export interface OrderItems{
    itemId: number,
    price: number,
    quantity: number,
    status: string
};

export interface Address{
    street: string,
    city: string,
    state: string,
    country: string
};